lib/index.js

```diff
@@ -438,6 +438,117 @@ Mongoose.prototype.modelNames = function() {
 Mongoose.prototype.modelNames.$hasSideEffects = true;
 
 /**
+ * Generates a new Schema merged from others
+ *
+ * @param {Object} [schema] Object with fields that the merged Schemas does not have
+ * @param {Schema|Array} schemas
+ * @param {Object} options options like in Schema constructor
+ * @param {Boolean} [control] says when the generated schema will have control over merged schemas
+ * @api public
+ * @return {Schema}
+ */
+Mongoose.prototype.mergeSchemas = function(schema, schemas, options, control) {
+  if (schema.constructor !== Object) {
+    control = options;
+    options = schemas;
+    schemas = schema;
+    schema = undefined;
+  }
+  schema = schema || {};
+
+  var dstSchema, dict = {};
+  var _kinds = [], _kind;
+  var ret; // return object
+
+  // control defaults to true
+  control = (control !== false);
```

Comment:
'control' is a boolean parameter that checks if we will create _kind and _kinds keys on the document before saving it.